src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.SkeletonReferenceCache.cs

```diff
@@ -78,54 +85,40 @@ private partial class SkeletonReferenceCache
         private static readonly ConditionalWeakTable<Compilation, AsyncLazy<SkeletonReferenceSet?>> s_compilationToSkeletonSet = new();
 
         /// <summary>
-        /// Lock around <see cref="_version"/> and <see cref="_skeletonReferenceSet"/> to ensure they are updated/read 
-        /// in an atomic fashion.
-        /// </summary>
-        private readonly object _stateGate = new();
-
-        /// <summary>
-        /// The <see cref="Project.GetDependentSemanticVersionAsync"/> version of the project that the
-        /// <see cref="_skeletonReferenceSet"/> corresponds to.
+        /// The <see cref="Project.GetDependentSemanticVersionAsync"/> version of the project that the <see
+        /// cref="_skeletonReferenceSet"/> corresponds to.  Initially set to <see cref="VersionStamp.Default"/>.
         /// </summary>
-        private VersionStamp? _version;
+        private VersionStamp _version;
 
         /// <summary>
         /// Mapping from metadata-reference-properties to the actual metadata reference for them.
         /// </summary>
         private SkeletonReferenceSet? _skeletonReferenceSet;
 
-        public SkeletonReferenceCache()
-            : this(version: null, skeletonReferenceSet: null)
-        {
-        }
-
-        private SkeletonReferenceCache(
-            VersionStamp? version,
-            SkeletonReferenceSet? skeletonReferenceSet)
-        {
-            _version = version;
-            _skeletonReferenceSet = skeletonReferenceSet;
-        }
```

Comment:
don't need/want the constructors.  The default state of this struct is correct and what we want.